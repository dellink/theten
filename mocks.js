export const tiles = [
  {
    value: 5,
    row: 0,
    column: 0,
    oldRow: -1,
    mergedInto: null,
    id: 'KQ317M0iPCG7dnn5HwBCK',
  },
  {
    value: 6,
    row: 0,
    column: 1,
    oldRow: -1,
    mergedInto: null,
    id: 'CeITToqKJ4hLt5XzfQuQA',
  },
  {
    value: 6,
    row: 0,
    column: 2,
    oldRow: -1,
    mergedInto: null,
    id: 'WI9fDkLCK4bi6aUH4SGi3',
  },
  {
    value: 3,
    row: 0,
    column: 3,
    oldRow: -1,
    mergedInto: null,
    id: 'bnYWFOluVTq0v28NIlNVU',
  },
  {
    value: 8,
    row: 1,
    column: 0,
    oldRow: -1,
    mergedInto: null,
    id: 'GMqOeXXVnqhxvIG1dONHp',
  },
  {
    value: 5,
    row: 1,
    column: 1,
    oldRow: -1,
    mergedInto: null,
    id: 'VWONtGPl8u7KwUw3QvOeB',
  },
  {
    value: 1,
    row: 1,
    column: 2,
    oldRow: -1,
    mergedInto: null,
    id: 'OqbcyZrr6RvzVHDFr81Nz',
  },
  {
    value: 7,
    row: 1,
    column: 3,
    oldRow: -1,
    mergedInto: null,
    id: 'kjxU8k_dTMPzfhga1e52Z',
  },
  {
    value: 7,
    row: 2,
    column: 0,
    oldRow: -1,
    mergedInto: null,
    id: 'Hx5pVa3tErUIKZfxGSheo',
  },
  {
    value: 4,
    row: 2,
    column: 1,
    oldRow: -1,
    mergedInto: null,
    id: '39OGk3pF13ydnZTR53JBR',
  },
  {
    value: 2,
    row: 2,
    column: 2,
    oldRow: -1,
    mergedInto: null,
    id: 'xWmRL7Z9GTwNcy8JItY70',
  },
  {
    value: 4,
    row: 2,
    column: 3,
    oldRow: -1,
    mergedInto: null,
    id: 'Gz9x2LpfV0YvjpAUpB6pe',
  },
  {
    value: 9,
    row: 3,
    column: 0,
    oldRow: -1,
    mergedInto: null,
    id: 'AQZEgW_RVS3BmFxzMkugJ',
  },
  {
    value: 1,
    row: 3,
    column: 1,
    oldRow: -1,
    mergedInto: null,
    id: 'S5p7c1sYTaUXzKfUYYAf5',
  },
  {
    value: 9,
    row: 3,
    column: 2,
    oldRow: -1,
    mergedInto: null,
    id: 'XKnfhJ3YpfNLrM-lDO1Dm',
  },
  {
    value: 2,
    row: 3,
    column: 3,
    oldRow: -1,
    mergedInto: null,
    id: 'TLVZ_tlfp-M1BALvwa6AQ',
  },
];
