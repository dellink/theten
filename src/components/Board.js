import React from 'react';
import { StyleSheet, View } from 'react-native';
import { BOARD_PADDING } from '../config';
import Cell from './Cell';
import Row from './Row';

export default function Board({ tiles, onMoveCell }) {
  const cells = tiles
    .filter(tile => tile.value)
    .map(tile => <Cell key={tile.id} tile={tile} onMoveCell={onMoveCell} />);

  return (
    <View style={styles.board}>
      <Row />
      <Row />
      <Row />
      <Row />
      {cells}
    </View>
  );
}

const styles = StyleSheet.create({
  board: {
    backgroundColor: '#03A9F4',
    borderColor: '#0288D1',
    borderRadius: 5,
    borderWidth: 1,
    padding: BOARD_PADDING,
  },
  row: {
    flexDirection: 'row',
  },
});
