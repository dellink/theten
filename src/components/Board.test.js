import React from 'react';
import { render } from '@testing-library/react-native';
import { tiles } from '../../mocks';
import Board from './Board';

jest.mock('./Cell', () => props => <div {...props} />);

describe(Board.name, () => {
  it('should render correctly', () => {
    expect(render(<Board tiles={tiles} />).asJSON()).toMatchSnapshot();
  });

  it('should render all cells', () => {
    const tileNodes = render(<Board tiles={tiles} />).container.findAll(node =>
      node.getProp('tile'),
    );
    expect(tileNodes).toHaveLength(tiles.length);
  });
});
