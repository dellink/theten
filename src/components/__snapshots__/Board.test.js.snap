// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#03A9F4",
        "borderColor": "#0288D1",
        "borderRadius": 5,
        "borderWidth": 1,
        "padding": 3,
      }
    }
  >
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
      <View
        style={
          Object {
            "backgroundColor": "#ddccbb",
            "borderRadius": 5,
            "height": 68,
            "margin": 5,
            "width": 68,
          }
        }
      />
    </View>
    <div
      tile={
        Object {
          "column": 0,
          "id": "KQ317M0iPCG7dnn5HwBCK",
          "mergedInto": null,
          "oldRow": -1,
          "row": 0,
          "value": 5,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 1,
          "id": "CeITToqKJ4hLt5XzfQuQA",
          "mergedInto": null,
          "oldRow": -1,
          "row": 0,
          "value": 6,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 2,
          "id": "WI9fDkLCK4bi6aUH4SGi3",
          "mergedInto": null,
          "oldRow": -1,
          "row": 0,
          "value": 6,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 3,
          "id": "bnYWFOluVTq0v28NIlNVU",
          "mergedInto": null,
          "oldRow": -1,
          "row": 0,
          "value": 3,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 0,
          "id": "GMqOeXXVnqhxvIG1dONHp",
          "mergedInto": null,
          "oldRow": -1,
          "row": 1,
          "value": 8,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 1,
          "id": "VWONtGPl8u7KwUw3QvOeB",
          "mergedInto": null,
          "oldRow": -1,
          "row": 1,
          "value": 5,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 2,
          "id": "OqbcyZrr6RvzVHDFr81Nz",
          "mergedInto": null,
          "oldRow": -1,
          "row": 1,
          "value": 1,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 3,
          "id": "kjxU8k_dTMPzfhga1e52Z",
          "mergedInto": null,
          "oldRow": -1,
          "row": 1,
          "value": 7,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 0,
          "id": "Hx5pVa3tErUIKZfxGSheo",
          "mergedInto": null,
          "oldRow": -1,
          "row": 2,
          "value": 7,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 1,
          "id": "39OGk3pF13ydnZTR53JBR",
          "mergedInto": null,
          "oldRow": -1,
          "row": 2,
          "value": 4,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 2,
          "id": "xWmRL7Z9GTwNcy8JItY70",
          "mergedInto": null,
          "oldRow": -1,
          "row": 2,
          "value": 2,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 3,
          "id": "Gz9x2LpfV0YvjpAUpB6pe",
          "mergedInto": null,
          "oldRow": -1,
          "row": 2,
          "value": 4,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 0,
          "id": "AQZEgW_RVS3BmFxzMkugJ",
          "mergedInto": null,
          "oldRow": -1,
          "row": 3,
          "value": 9,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 1,
          "id": "S5p7c1sYTaUXzKfUYYAf5",
          "mergedInto": null,
          "oldRow": -1,
          "row": 3,
          "value": 1,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 2,
          "id": "XKnfhJ3YpfNLrM-lDO1Dm",
          "mergedInto": null,
          "oldRow": -1,
          "row": 3,
          "value": 9,
        }
      }
    />
    <div
      tile={
        Object {
          "column": 3,
          "id": "TLVZ_tlfp-M1BALvwa6AQ",
          "mergedInto": null,
          "oldRow": -1,
          "row": 3,
          "value": 2,
        }
      }
    />
  </View>
</View>
`;
